examples/plot_callosal_tract_profile.py

```diff
@@ -0,0 +1,558 @@
+                  'params': op.join(working_dir, 'dti_params.nii.gz')}
+
+print(f"Loading {dti_params['FA']}")
+FA_img = nib.load(dti_params['FA'])
+FA_data = FA_img.get_fdata()
+
+show_anatomical_slices(FA_data, 'Fractional Anisotropy (FA)')
+
+##########################################################################
+# Register the individual data to a template:
+# -------------------------------------------
+# For the purpose of bundle segmentation, the individual brain is registered
+# to the MNI T2 template. The waypoint ROIs used in segmentation are then each
+# brought into each subject's native space to test streamlines for whether they
+# fulfill the segmentation criteria.
+#
+# .. note::
+#    Here, for simplicity we only perform a non-linear transformation between
+#    the individual's brain and the MNI T2 template. This is likely why the
+#    occipital/parietal junction appears concave. In practice, it's a good idea
+#    to also perform a pre-alignment using an affine transformation. Moreover,
+#    by default, registration uses APM (Anisotropic Power Map) which also tends
+#    to provide better results.
+
+
+print("Registering to template...")
+
+MNI_T2_img = afd.read_mni_template()
+nib.save(MNI_T2_img, op.join(working_dir, 'MNI_T2.nii.gz'))
+show_anatomical_slices(MNI_T2_img.get_fdata(), 'MNI T2 Template')
+
+# create a diffeomorphic map from subject to MNI space
+if not op.exists(op.join(working_dir, 'mapping.nii.gz')):
+    import dipy.core.gradients as dpg
+    gtab = dpg.gradient_table(hardi_fbval, hardi_fbvec)
+
+    warped_hardi, mapping = reg.syn_register_dwi(hardi_fdata, gtab)
+    reg.write_mapping(mapping, op.join(working_dir, 'mapping.nii.gz'))
+else:
+    mapping = reg.read_mapping(op.join(working_dir, 'mapping.nii.gz'),
+                               img,
+                               MNI_T2_img)
+
+mapping_img = nib.load(op.join(working_dir, 'mapping.nii.gz'))
+mapping_img_data = mapping_img.get_fdata()
+
+# Working with diffeomorphic map data
+mapping_img_data = mapping_img_data[..., 0, 0]
+show_anatomical_slices(mapping_img_data, 'Registration Displacement Mapping')
+
+# plot the transformational map of MNI T2 onto subject space
+warped_MNI_T2_data = mapping.transform_inverse(MNI_T2_img.get_fdata())
+warped_MNI_T2_img = nib.Nifti1Image(warped_MNI_T2_data.astype(float),
+                                    img.affine)
+
+nib.save(warped_MNI_T2_img, op.join(working_dir, 'warped_MNI_T2.nii.gz'))
+
+show_anatomical_slices(warped_MNI_T2_img.get_fdata(), 'Warped MNI T2')
+
+##########################################################################
+# Create novel bundle specification:
+# ----------------------------------
+# The bundles specify meta-data for the segmentation.
+#
+# The following keys are required for a `bundle` entry:
+#
+# - `ROIs`
+#
+#   lists the ROI templates.
+#
+# .. note::
+#   Order of the `ROIs` matters and may result in different tract profiles
```

Comment:
@arokem I know we discussed why order matters, but I didn't follow closely enough to provide details here

I tested flipping the left and right ROIs and returned similar but different streamlines (was motivation for the diff request above)

Reply:
Why order sequence of ROIs in bundle matters:
Given sequence of ROIs: (e.g., A, B, C)
The waypoint ROIs endpoints should be first two (i.e., A, C, B)
Note that directionality doesn't matter, (i.e. A <-> C: A ->C or C -> A); however want to be consistent with with other streamlines

NOTE: aal endpoints interact as well, but we side step in this example since not using endpoints